extensions/sql-migration/src/wizard/targetSelectionPage.ts

```diff
@@ -89,7 +90,16 @@ export class TargetSelectionPage extends MigrationWizardPage {
 				this._azureResourceDropdown.ariaLabel = constants.AZURE_SQL_DATABASE_VIRTUAL_MACHINE;
 				break;
 		}
+
 		await this.populateAzureAccountsDropdown();
+		if (this._migrationTargetPlatform !== this.migrationStateModel._targetType) {
+			// if the user had previously selected values on this page, then went back to change the migration target platform
+			// and came back, forcibly reload the location/resource group/resource values since they will now be different
+			this._migrationTargetPlatform = this.migrationStateModel._targetType;
+			await this.populateLocationDropdown();
+			await this.populateResourceGroupDropdown();
+			await this.populateResourceInstanceDropdown();
```

Comment:
should these methods calls be removed, I think you are filling the dropdowns multiple times.
			await this.populateResourceGroupDropdown();
			await this.populateResourceInstanceDropdown();
where calling await this.populateLocationDropdown(); 
should trigger calling await this.populateResourceGroupDropdown();
and then trigger calling await this.populateResourceInstanceDropdown();

Reply:
Generally speaking yes. I had removed such calls from previous iterations of this PR. This one specifically is the only place I do this now. The only reason is that if the user goes back from the target selection page, changes target platforms and comes back, I need to forcibly clear all the dropdown values at once (see my comment in code above). But if they didn't change their target platform, the page should remember the previous values so that the user doesn't have to reenter them all. This wasn't a problem in the past because the filtering logic was different at the time